### PR TITLE
Refactor Json

### PR DESCRIPTION
**Pull request by bmdhacks**
Ran down a side-path and refactored the json generation code.  This is a first step with no removal of existing functionality, but it makes it much easier to start removing the Json object conversions.

The main architectural change is to use a Jackson generator object to do the type conversion instead of scala type-pimping and the .toJson method.

Unit tests aren't rewritten, but I wanted to put this up to get feedback before pursuing further.

If this looks good, the next step is to start removing JsonXXX objects and to make the actual classes convert natively.

### CODE REVIEW

File: zipkin-web/src/main/scala/com/twitter/zipkin/common/json/ZipkinJson.scala [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package com.twitter.zipkin.common.json
+
+import com.codahale.jerkson.Json
+import org.codehaus.jackson.map.annotate.JsonCachable
+import com.twitter.zipkin.common.BinaryAnnotation
+import com.fasterxml.jackson.databind.{SerializerProvider, JsonSerializer}
+import com.fasterxml.jackson.core.{JsonGenerator=>JacksonGenerator}
+import com.fasterxml.jackson.databind.module.SimpleModule
+
```
Comment (franklinhu) [inline zipkin-web/src/main/scala/com/twitter/zipkin/common/json/ZipkinJson.scala]: whitespace

Comment (bmdhacks) [inline zipkin-web/src/main/scala/com/twitter/zipkin/common/json/ZipkinJson.scala]: do what now?
http://www.youtube.com/watch?v=h06cSTAqAYk

**Comment by franklinhu** [comment]
lgtm

**Comment by bmdhacks** [comment]
Gah, did the wrong operation.  Gonna squash-merge by hand. :-p

**Pull request closed without merging**